packages/SwingSet/tools/run-utils.ts

```diff
@@ -0,0 +1,152 @@
+  // normal code.  However, this convenience comes with a huge caveat that all
+  // users of this convenience feature MUST keep in mind.
+  //
+  // A test can drop messages onto the kernel's run queue using the
+  // `controller.queueToVatRoot` and `controller.queueToVatObject` methods.
+  // These are synchronous operations which merely place messages onto the run
+  // queue without causing execution.  Execution, on the other hand, is
+  // initiated by calling `controller.run`, which will cause the kernel to begin
+  // delivering messages to vats from the run queue, continuing until the run
+  // queue is exhausted.  HOWEVER, exhaustion of the run queue, which resolves
+  // the result promise returned by the `run` call, IS NOT coupled in any causal
+  // way to the resolution of result promises associated with the individual
+  // queued messages themselves.  The status and resolution values of these
+  // promises can be synchronously queried (by kpid) via the
+  // `controller.kpStatus` and `controller.kpResolution` methods once `run` has
+  // completed.  These queries are only available once the swingset has
+  // reqlinquished agency, i.e., when the work initiated by `controller.run` has
+  // finished.  At that point, nothing is going on inside the kernel, and
+  // nothing WILL be going on inside the kernel until a subsequent call to
+  // `controller.run`, which in turn will only have an effect if additional
+  // messages have been placed on the kernel run queue in the meantime.  You MAY
+  // NOT call `queueToVatRoot`, `queueToVatObject`, `kpStatus`, or
+  // `kpResolution` while run queue execution, triggered by a call to `run`, is
+  // in progress
+  //
+  // The functionality made available by `EV` looks similar to that provided by
+  // `E`, but it is very much not.  When you send a message using `EV`, it
+  // places the message onto the kernal run queue and then immediately invokes
+  // `controller.run`.  When the result of `run` resolves, the kpid returned by
+  // the message enqueueing operation is queried.  If at that time the promise
+  // it identifies is resolved (or rejected), the value it was resolved (or
+  // rejected) to is used as the result from the `EV` invocation.  However, if
+  // it is still pending at that time, `EV` will throw an exception and your
+  // test will abort.  This means that if you initiate some operation via an
+  // `EV` message send, it must complete within a single `run` cycle for it to
+  // be of any use to you.  This is quite different from a message sent using
+  // `E`, which will return a promise that can remain pending indefinitely,
+  // possibly to be resolved by a future message delivery.
+
+  // @ts-expect-error cast, approximate
+  const EV: EVProxy = presence =>
+    new Proxy(harden({}), {
+      get: (_t, method, _rx) =>
+        harden((...args) =>
+          queueAndRun(() =>
+            controller.queueToVatObject(presence, method, args),
+          ),
+        ),
```

Comment:
Comprehensibility suggestion:
```suggestion
      get: (_t, methodName, _rx) => {
        const boundMethod = (...args) => queueAndRun(() =>
          controller.queueToVatObject(presence, methodName, args)
        );
        return harden(boundMethod);
      },
```

Reply:
I was just using what was there before I touched this code, but this really does look like an improvement to me.  I personally find the arrow function syntax confusing under the best of circumstances, but arrow functions returning expressions containing arrow functions are even more so (and here we have three levels of nesting, which is just terrible).  I find FP-style code is often prone to performative cleverness, but not in a fun way like Kris or MarkM's puns.